graphql/student/mutations.ts

```diff
@@ -160,7 +202,73 @@ export async function updateStudent(context: GraphQLContext, student: Student, u
     await updateSessionUser(context, userForStudent(res));
 
     log.info(`Student(${student.id}) updated their account with ${JSON.stringify(update)}`);
+    return res;
+}
+
+async function studentRegisterPlus(data: StudentRegisterPlusInput, ctx: GraphQLContext): Promise<{ success: boolean; reason: string }> {
+    const log = logInContext('Student', ctx);
+    const { email, register, activate, screen } = data;
+    const screener = await getSessionScreener(ctx);
+
+    try {
+        if (register && register.email !== email) {
+            throw new PrerequisiteError(`Identifying email is different from email used in registration data`);
+        }
+
+        const existingAccount = await prisma.student.findUnique({ where: { email } });
+
+        if (!register && !existingAccount) {
+            throw new PrerequisiteError(`Account with email ${email} doesn't exist and no registration data was provided`);
+        }
+
+        await prisma.$transaction(async (tx) => {
+            let student;
+            if (!!register) {
+                //registration data was provided
+                if (!!existingAccount) {
+                    log.info(`Account with email ${email} already exists, updating account with registration data instead...`);
+                    // updating existing account with new registration data:
+                    student = await updateStudent(ctx, existingAccount, { ...register, projectFields: undefined, languages: undefined }, tx); // languages are added in next step (becomeTutor)
+                } else {
+                    student = await registerStudent(register, true, tx);
+                    log.info(`Registered account with email ${email}`);
```

Comment:
I have the fear that sooner or later something goes wrong during a mass import, and then we have to search for broken accounts. For that case we should have good logging, so I'd suggest to:
- Log the start and end of the mass import (so that one can filter for the range)
- Add Pupil({id}) and Student({id}) to every log line so that one can filter by those too

Apart from that, nice PR!

Reply:
I thought the same while reading the code, but Marius already took care of most cases with using transactions. 🎉 

Do you have any thoughts about what still can go wrong?

Reply:
Not really, but I guess the whole thing might get more complex in the future.

Maybe alternatively we could just write the result into the log? So in case someone runs the thing in the Retool but forgets to store the output, we still know what they saw.